samples/matter/lock/CMakeLists.txt

```diff
@@ -16,13 +16,12 @@ if(CONF_FILE)
     get_filename_component(CONF_FILE_NAME ${CONF_FILE} NAME CACHE)
 endif()
 
-if(NOT CONF_FILE_NAME STREQUAL "prj_no_dfu.conf")
-    set(PM_STATIC_YML_FILE ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/pm_static_dfu.yml)
-endif()
-
 if(CONF_FILE_NAME STREQUAL "prj_thread_wifi_switched.conf")
+    set(PM_STATIC_YML_FILE ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/pm_static_dfu_wifi_ext_patch.yml)
```

Comment:
@tejlmand

Reply:
@tejlmand it seems that we and many other teams do it for a while. As this PR does not change the way how things are handled I would like to ask to address it after NCS 2.6.0 as those are not related to this PR directly. And we really want this PR to get in before code freeze.

We do understand that by making tens of duplicated files and content inside, we can achieve the same - but we really don't want to go this path as we failed to maintain those in the past.

Reply:
yes, part of this gets down to a more fundamental question on general use.

Also because the `(NOT CONF_FILE_NAME STREQUAL "prj_no_dfu.conf")` below breaks the built-in support described here:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html#configuring-static-partitions

which then results in the `if(CONF_FILE_NAME STREQUAL "prj_thread_wifi_switched.conf")` test.
A test which is now needed because of the lines below.